llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -601,7 +601,7 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
   auto setF16Action = [&] (MVT VT, LegalizeAction Action) {
     setOperationAction(ISD::FABS, VT, Action);
     setOperationAction(ISD::FNEG, VT, Action);
-    setOperationAction(ISD::FCOPYSIGN, VT, Expand);
+    setOperationAction(ISD::FCOPYSIGN, VT, Action);
```

Comment:
What's the reason to change this? And why don't set above Expand here?

Reply:
Because it would create even more special cases that don't reflect the function name.

The function is called elsewhere with `Expand` unless it's `f16`.

That's just for consistency, I could do it here if it's better.

Reply:
This function is called not only by scalar `f16` but also both vector FP16 and BF16, though we don't have tests to cover them.

Reply:
When it is for a vector FP/BF, it is always called with the Expand Operation.

Reply:
Thanks, then it's fine for me :)

Reply:
Wait.. should we remove it from `setF16Action` instead?

Reply:
> Wait.. should we remove it from `setF16Action` instead?

Sorry, I misunderstood. No problem then.

Reply:
1. we can isolate those Opcodes into another function.
2. keep them and overwrite the actions just after calling `setF16Action` if that's not the right ones (which is the path I've taken, and that's what's in the current code with `MVT::v16f16` for example).

Reply:
I was thinking you meant `FCOPYSIGN` is set Expand by default, then I got you mean it's all set Expand in X86ISelLowering.cpp.
Just have a double check, it does set Expand by default, see
https://github.com/llvm/llvm-project/blob/main/llvm/lib/CodeGen/TargetLoweringBase.cpp#L775
So we can simply remove it from `setF16Action`.

Reply:
That's right. The less code the better.